docs/static/generated/rest_v1_dispatcher.yml

```diff
@@ -1089,6 +1089,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/JobVertexBackPressureInfo'
+  /jobs/{jobid}/vertices/{vertexid}/coordinator-metrics:
+    get:
+      description: Provides access to job manager operator metrics
```

Comment:
Looking further into the implementation, it seems that it would make sense to have a broader `getJobManagerOperatorMetrics` method in case we expect other kind of JM operators to exist (not just the coordinators), and add a `getCoordinatorMetrics` wrapper to filter the scope of the former.

Reply:
Sounds good. I'll add a short description. As for the method change suggestion, I'd rather keep the scope limited. It caused confusion in my implementation and it took some digging to figure out that the there is only one coordinator, at least in the Flink Public API. As it is an internal method, someone can easily change this down the road when the requirement surfaces

Reply:
My concern is that someone adds another other type of the JM operator to the results returned by `getJobManagerOperatorMetrics` and we start to accidentally displaying it as a coordinator, despite it not being one.

Reply:
I could add filtering logic on the endpoint but should we even do filtering? In the future, if we extend JobManagerOperator to not just coordinator metrics, do we want to display it on the UI? 

My thought is yes and if that's the case, we probably should go back to the original proposal to keep the endpoint to not be specific to coordinators.

Reply:
Maybe I am missing something, but should not the endpoint called `coordinator-metrics` only return metrics for coordinators?

Reply:
Yes it should, but my question is more about the scope of the endpoint and this feature.

If a user wants other JM operator metrics in the future, what would happen?
1. We would need to support another endpoint (possibly deprecate this one and create a new one that supports all JM operator metrics)
2. Integrate with the Flink UI again

Also, the feedback from @zentol and @lindong28 in [1] reflects that the devs were against `coordinator-metrics` and introducing a specific scope to support that, and perhaps that feedback also applies to this REST API. I had this as my original proposal in the voting thread for the FLIP to support `/jobs/.../vertices/.../jm-operator-metrics`

[1] https://lists.apache.org/thread/hdvz57wmlqxdxgnw73tcgbcfyo6vk09h

Reply:
I see, thanks for the clarification. I believe the main issue with the original proposal was that it also implied that the user would need to supply operator ID (as reflected in the FLIP's rejected approaches `/jobs/<jobid>/vertices/<vertexid>/operators/<operatorid>/metrics`). This would necessitate an additional step to identify which operator serves as the coordinator.

It seems the challenge of distinguishing between the coordinator's metrics and other types of JobManager operators that may emerge in the future remains.  Suppose we consolidate everything under the `/jm-operator-metrics` endpoint. When focusing on the coordinator's metrics for autoscaling purposes, how will API users distinguish these from other metrics retrieved from `/jm-operator-metrics`? Can be sure that the metrics of interest are always uniquely identified by their names, preventing any overlap with those emitted by other operators?

Reply:
Yes, introducing the operator id would make it difficult for the UI to integrate since metrics APIs rely on the vertexid at the lowest level of granularity.

>When focusing on the coordinator's metrics for autoscaling purposes, how will API users distinguish these from other metrics retrieved from /jm-operator-metrics? Can be sure that the metrics of interest are always uniquely identified by their names, preventing any overlap with those emitted by other operators?
Ah yes, thanks for the suggestion! It would be possible if we formatted the metric by `<operator-name>.<metric-name>`, which I already do in the PR!